tests/components/blebox/test_light.py

```diff
@@ -0,0 +1,597 @@
+
+    state = hass.states.get(entity_id)
+    assert state.name == "dimmerBox-brightness"
+
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert supported_features & SUPPORT_BRIGHTNESS
+
+    assert state.attributes[ATTR_BRIGHTNESS] == 65
+    assert state.state == STATE_ON
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My dimmer"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "dimmerBox"
+    assert device.sw_version == "1.23"
+
+
+async def test_dimmer_update(dimmer, hass, config):
+    """Test light updating."""
+
+    feature_mock, entity_id = dimmer
+
+    def initial_update():
+        feature_mock.brightness = 53
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.attributes[ATTR_BRIGHTNESS] == 53
+    assert state.state == STATE_ON
+
+
+async def test_dimmer_on(dimmer, hass, config):
+    """Test light on."""
+
+    feature_mock, entity_id = dimmer
+
+    def initial_update():
+        feature_mock.is_on = False
+        feature_mock.brightness = 0  # off
+        feature_mock.sensible_on_value = 254
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+    def turn_on(brightness):
+        assert brightness == 254
+        feature_mock.brightness = 254  # on
+        feature_mock.is_on = True  # on
+
+    feature_mock.async_on = AsyncMock(side_effect=turn_on)
+    await hass.services.async_call(
+        "light", SERVICE_TURN_ON, {"entity_id": entity_id}, blocking=True,
+    )
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_BRIGHTNESS] == 254
+
+
+async def test_dimmer_on_with_brightness(dimmer, hass, config):
+    """Test light on with a brightness value."""
+
+    feature_mock, entity_id = dimmer
+
+    def initial_update():
+        feature_mock.is_on = False
+        feature_mock.brightness = 0  # off
+        feature_mock.sensible_on_value = 254
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+    def turn_on(brightness):
+        assert brightness == 202
+        feature_mock.brightness = 202  # on
+        feature_mock.is_on = True  # on
+
+    feature_mock.async_on = AsyncMock(side_effect=turn_on)
+
+    def apply(value, brightness):
+        assert value == 254
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)